### ISSUE TITLE
IDE0004 on pointer cast in return statement

### ISSUE THREAD

**Issue by NextTurn** [dotnet/roslyn#38599]
**Version Used**: Visual Studio 2019 version 16.3

**Steps to Reproduce**:

```cs
using System;

class Program
{
    public static int Value;

    public static unsafe int Read(IntPtr pointer, int offset)
    {
        Value = ((int*)pointer)[offset]; // No warning
        return ((int*)pointer)[offset];
        //      ~~~~~~
    }
}
```

**Expected Behavior**:

IDE0004 is not reported for the cast.

**Actual Behavior**:

IDE0004 is reported for the cast.

**Comment by sharwell** [comment]
Duplicate of #25021
